src/inject/settings.store.ts

```diff
@@ -43,50 +41,29 @@ export class SettingsStore
 
 	pullSettings(): Promise<any>
 	{
-		return new Promise( ( resolve, reject ) =>
+		return new Promise( ( resolve ) =>
 		{			
-			const callback = values => 
-			{
-				// If we are running this for a first time, there might be
-				// some values undefined. So we try to merge these that are not
-				// undefined with a previous state (in a first time it's default values).
-
-				const newState = Object.assign( {}, this.state );
-				
-				Object.keys( values ).forEach( key =>
-				{
-					if ( values.hasOwnProperty( key ) )
-					{
-						const value = values[ key ];
-
-						if ( value !== undefined || value !== null )
-						{
-							newState[ key ] = value;
-						}
-					}
-				})
-
-				this.setState( newState )
-				resolve( newState )
-			}
-
-			
-			// Chrome uses callbacks, Firefox uses Promises, so this is cross-browser solution
-			const keys = Object.keys( this.defaultValues );
-			const storage = this.storage.get( keys, callback )
-			
-			if ( storage && storage.then )
-			{
-				storage.then( callback )
-			}
+			const result = Object.assign( {}, this.state )
+
+			Object.keys(this.defaultValues).forEach((key) => {
+				const val = this.storage.getItem(key)
+				val && (result[key] = val)
+			})
```

Comment:
As you have no side effects you could use reduce here:

`const result = Object.keys(this.defaultValues).reduce((prev, key) => {
   const val = this.storage.getItem(key);
   if (val) {
       prev[key] = val;
   }
   return prev;
}, { ...this.state });`